actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -51,7 +51,7 @@ def collection_by_cache_keys
       end
 
       def expanded_cache_key(key)
-        key = @view.fragment_cache_key(@view.cache_fragment_name(key))
+        key = @view.fragment_cache_key(@view.cache_fragment_name(key, virtual_path: @template.virtual_path))
```

Comment:
@repinel do you have a good understanding of why `@view`'s `virtual_path` is nil, when this is called? I've forgotten parts of this code, so do fill me in if you know :smile: